File: playschool_app/lib/models/fee.dart
class Fee {
  final String id;
  final String childName;
  final String month;
  final int monthlyFee;
  final int paidAmount;
  final String status;
  final String paymentDate;

  Fee({
    required this.id,
    required this.childName,
    required this.month,
    required this.monthlyFee,
    required this.paidAmount,
    required this.status,
    required this.paymentDate,
  });

  factory Fee.fromJson(
      Map<String, dynamic> json) {
    return Fee(
      id: json['_id'],
      childName:
          json['childId']['name'],
      month: json['month'],
      monthlyFee:
          json['monthlyFee'],
      paidAmount:
          json['paidAmount'],
      status: json['status'],
      paymentDate:
          json['paymentDate'] ?? '',
    );
  }
}

File: playschool_backend/models/Fee.js
const mongoose = require("mongoose");

const feeSchema = new mongoose.Schema(
  {
    childId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Child",
      required: true,
    },

    month: {
      type: String,
      required: true,
    },

    monthlyFee: {
      type: Number,
      required: true,
    },

    paidAmount: {
      type: Number,
      default: 0,
    },

    status: {
      type: String,
      enum: ["Paid", "Pending"],
      default: "Pending",
    },

    paymentDate: {
      type: String,
      default: "",
    },

    paymentMethod: {
      type: String,
      default: "",
    },
  },
  {
    timestamps: true,
  }
);

feeSchema.index(
  {
    childId: 1,
    month: 1,
  },
  {
    unique: true,
  }
);

module.exports =
  mongoose.model(
    "Fee",
    feeSchema
  );